core/domain/user_services.py

```diff
@@ -901,46 +544,18 @@ def get_user_ids_by_role(role):
     return [user.id for user in user_settings]
 
 
-class UserActionsInfo(python_utils.OBJECT):
-    """A class representing information of user actions.
-
-    Attributes:
-        user_id: str. The unique ID of the user.
-        role: str. The role ID of the user.
-        actions: list(str). A list of actions accessible to the role.
-    """
-
-    def __init__(self, user_id=None):
-        self._user_id = user_id
-        self._role = get_user_role_from_id(user_id)
-        self._actions = role_services.get_all_actions(self._role)
-
-    @property
-    def user_id(self):
-        """Returns the unique ID of the user.
+def get_user_actions_info(user_id):
+    """Get user actions info for a user.
 
-        Returns:
-            user_id: str. The unique ID of the user.
-        """
-        return self._user_id
-
-    @property
-    def role(self):
-        """Returns the role ID of user.
-
-        Returns:
-            role: str. The role ID of the user.
-        """
-        return self._role
-
-    @property
-    def actions(self):
-        """Returns list of actions accessible to a user.
+    Args:
+        user_id: str|None. The user ID of the user we want to get actions for.
 
-        Returns:
-            actions: list(str). List of actions accessible to a user ID.
-        """
-        return self._actions
+    Returns:
+        UserActionsInfo. User object with system committer user id.
+    """
+    role = get_user_role_from_id(user_id)
+    actions = role_services.get_all_actions(role)
+    return user_domain.UserActionsInfo(user_id, role, actions)
```

Comment:
Nice improvement: it's always codesmell to perform potentially expensive operations in class constructors--I'm glad to see this refactor.